### PR TITLE
Added usage of datastore to save discovery cache

### PR DESCRIPTION
**Pull request by alon-tchelet**
Samsung told us that in an edge case in which the hub or the device crashes during discovery stages the device could be initialised improperly. They informed us about how to try to reduce the damage

### CODE REVIEW

**Review by dljsjr — APPROVED** [review]

**Comment by lelandblue** [comment]
Hello @alon-tchelet What are the next steps on this PR? Have these changes been tested?

**Comment by alon-tchelet** [comment]
> Hello @alon-tchelet What are the next steps on this PR? Have these changes been tested?

The AVR series added haven't got to the market yet and will go into certification in the coming month or two. So was only tested in dev environemnts.
As for the driver updates and modifications for the L42ms, they are tested by the Harman led testing teams

**Comment by alon-tchelet** [comment]
After contact from Samsung, we reverted the new product changes and I will make a separate PR for introducing the new devices (new profile) as they suggested

**Comment by Seong-Yeol-Park** [comment]
Hello @alon-tchelet, 
If there is no issue in your self test, could you ask @lelandblue to merge this PR?
And then WWST team can proceed CbS for new MA series in #1149 .

**Comment by alon-tchelet** [comment]
Hi @Seong-Yeol-Park,
I agree. This could (and should) be merge now.

**Comment by varzac** [comment]
This PR has merge conflicts with the code that is already in the repository.  Those will have to be resolved before we can merge this.

**Comment by alon-tchelet** [comment]
Hi @varzac ,

I had a look and as PR #1149 included the changes from this PR, there is no need to merge this one too. Ideally this PR should have been reviewed and merged before PR #1149 . But as the other one was merged first, conflicts appeared.

I don't know what is your process for this case, but this PR is no longer needed and its content is already merged. It could be just closed.

**Comment by cjswedes** [comment]
> I had a look and as PR https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1149 included the changes from this PR, there is no need to merge this one too. Ideally this PR should have been reviewed and merged before PR https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1149 . But as the other one was merged first, conflicts appeared.
I don't know what is your process for this case, but this PR is no longer needed and its content is already merged. It could be just closed.

Feel free to just close this PR. There is one outstanding change after the merge of upstream/main; its not clear if that is still needed.

**Comment by alon-tchelet** [comment]
Hi @cjswedes ,
Can you point this change to me? I just want to make sure it's indeed not needed before closing this PR

File: drivers/SmartThings/harman-luxury/profiles/maX10.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ components:
   - id: refresh
     version: 1
   categories:
-  - name: Speaker
+  - name: Receiver
```
Comment (cjswedes) [inline drivers/SmartThings/harman-luxury/profiles/maX10.yaml:26]: This is the change.

Comment (alon-tchelet) [reply drivers/SmartThings/harman-luxury/profiles/maX10.yaml:26]: Oh I forgot that this PR is grubbing all my pushed commits.
This was actually part of a new commit - PR #1357 

So I will close this PR now

**Pull request closed without merging**